src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.cs

```diff
@@ -201,7 +211,21 @@ async Task SyncPlatformModalStackAsync()
 			if (FireLifeCycleEvents)
 			{
 				modal.SendNavigatingFrom(new NavigatingFromEventArgs());
-				modal.SendDisappearing();
+			}
+
+			modal.SendDisappearing();
+
+			// With shell we want to make sure to only fire the appearing event
+			// on the final page that will be visible after the pop has completed
+			if (_window.Page is Shell shell)
+			{
+				if (!shell.CurrentItem.CurrentItem.IsPoppingModalStack)
+				{
+					CurrentPage?.SendAppearing();
```

Comment:
Don't we need `previousPage?.SendDisappearing();`  ?

Reply:
That's what line 216 is
https://github.com/dotnet/maui/pull/15799/files#diff-a02814d4c61d2347359752528537440308b21bb97f59d85a05bdd711553c149dR216

it's calling disappearing on the modal that's being popped off the stack